sdk/eventhub/azure-eventhubs/azure/eventhub/aio/_consumer_client_async.py

```diff
@@ -0,0 +1,187 @@
+        """
+        :param host: The hostname of the Event Hub.
+        :type host: str
+        :param event_hub_path: The path of the specific Event Hub to connect the client to.
+        :type event_hub_path: str
+        :param credential: The credential object used for authentication which implements particular interface
+         of getting tokens. It accepts ~azure.eventhub.EventHubSharedKeyCredential,
+         ~azure.eventhub.EventHubSASTokenCredential, credential objects generated by the azure-identity library and
+         objects that implement get_token(self, *scopes) method.
+        :keyword network_tracing: Whether to output network trace logs to the logger. Default
+         is `False`.
+        : network_tracing: bool
+        :param http_proxy: HTTP proxy settings. This must be a dictionary with the following
+         keys: 'proxy_hostname' (str value) and 'proxy_port' (int value).
+         Additionally the following keys may also be present: 'username', 'password'.
+        :type http_proxy: dict[str, Any]
+        :param auth_timeout: The time in seconds to wait for a token to be authorized by the service.
+         The default value is 60 seconds. If set to 0, no timeout will be enforced from the client.
+        :type auth_timeout: float
+        :param user_agent: The user agent that needs to be appended to the built in user agent string.
+        :type user_agent: str
+        :param retry_total: The total number of attempts to redo the failed operation when an error happened. Default
+         value is 3.
+        :type retry_total: int
+        :param transport_type: The type of transport protocol that will be used for communicating with
+         the Event Hubs service. Default is ~azure.eventhub.TransportType.Amqp.
+        :type transport_type: ~azure.eventhub.TransportType
+        :param partition_manager: stores the load balancing data and checkpoint data when receiving events
+         if partition_manager is specified. If it's None, this EventHubConsumerClient instance will receive
+         events without load balancing and checkpointing.
+        :type partition_manager: PartitionManager.
+        :param load_balancing_interval: When load balancing kicks in, this is the interval in seconds
+         between two load balancing. Default 10.
+        :type load_balancing_interval: float
+        """
+
+        super(EventHubConsumerClient, self).__init__(
+            host=host, event_hub_path=event_hub_path, credential=credential, **kwargs)
+        self._partition_manager = kwargs.get("partition_manager")
+        self._load_balancing_interval = kwargs.get("load_balancing_interval", 10)
+        self._event_processors = dict()  # type: Dict[Tuple[str, str], EventProcessor]
+        self._closed = False
+
+    async def receive(
+            self, on_event: Callable[[PartitionContext, List[EventData]], None], consumer_group: str,
+            *,
+            partition_id: str = None,
```

Comment:
Why not default `partition_id` to -1? You are swapping between `None` as a parameter value being the same as `-1` in the consuming code anyways.

Reply:
I think "None" is more meaningful to the users. It means no partition_id. "-1" isn't a real partition_id. Better not expose it to users. Internally I use "-1" to denote all partitions

Reply:
I don't think that's a good idea to use a random string with no meaning repeatudly in the code. Use a constant, where value can't conflict. Example, assuming -1 can't be a partition_id
```python
ALL_PARTITIONS = "-1"

...

self._event_processors[(consumer_group, ALL_PARTITIONS)]
...
if (consumer_group, ALL_PARTITIONS) in self._event_processors:
```

If that's a magic string, it should be a constant

Reply:
Not blocking for preview5, solving